src/toolkits/image_deep_feature_extractor/mlmodel_image_feature_extractor.mm

```diff
@@ -162,10 +162,33 @@ void build_vision_feature_print_screen_spec(const std::string& model_path) {
 static MLModel *create_model(const std::string& download_path,
 			     const std::string& model_name) {
 
+  @autoreleasepool {
+
+  MLModel* result = nil;
+
   const std::string compiled_modified_model_path = download_path + "/" + model_name + "_modified.mlmodelc";
+  NSURL* compiledModelURL = [NSURL fileURLWithPath:[NSString stringWithUTF8String:compiled_modified_model_path.c_str()]];
+
+  // If the compiled modified model already exists on disk, attempt to load it.
+  if (boost::filesystem::exists(compiled_modified_model_path)) {
+
+    NSError* error = nil;
+    result = [MLModel modelWithContentsOfURL:compiledModelURL error:&error];
+
+    if (error) {
```

Comment:
Do you want to check `result` here too?

Reply:
Good call! Once upon a time, Cocoa APIs had a standard behavior of returning nil for exactly one of the return value and the NSError out parameter, but I'm not sure this is really an invariant anymore. We may as well define success here as _both_ `error` being nil and `result` being non-nil.